### PR TITLE
Changed maracas

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
(Has sound)

[maraca.webm](https://user-images.githubusercontent.com/41342767/190183497-69dc7bf0-32d4-4efb-922d-d47604df44ce.webm)

(The music keeps playing a little after the video)

Maracas can no longer be emagged to unlock their explosive functionality and now have a delay of about 2 seconds before it detonates, during which it plays a familiar tune. However, their explosion power has been DOUBLED, making them vastly more powerful.
A friendly reminder that you can get explosive maracas from beating Cuban Pete in the Outbomb Cuban Pete arcade game.

:cl:
 * tweak: Maracas can no longer be emagged.
 * soundadd: Maracas now play a familiar tune before exploding.
 * tweak: The explosive power of explosive maracas has been drastically increased.

### CODE REVIEW

**Comment by west3436** [comment]
based emag anti-creep proponent

**Comment by Eneocho** [comment]
cringe emag anti-creep soyman

**Comment by ghost** [comment]
honestly they shouldn't explode emag or not

**Comment by PrimeDSS13** [comment]
> honestly they shouldn't explode emag or not

They're the reward from the Cuban Pete emagged arcade game, which is incredibly hard to win and will more likely kill you.
I think it's fine if it's the only way to obtain the explosive version, unlike before.

**Comment by Kurfursten** [comment]
I actually like this. Emagging an arcade which then dispenses a dangerous reward makes more sense than rubbing your card on a piece of wood and suddenly it's a bomb. Emags should be less "outright magic" and more "hacker known as 4chan". Remember the height of emag fervor whens someone wanted to add emagging to paper?

And for the record, I'm **not** against features for the emag even though it already has uses. For example, if someone wanted to make it so the microwave can be emagged to always ruin food and hurl the burned mess into mouths a-la pneumo cannon, or emag hydro trays so they display wrong warning lights, or emag the chem dispenser to change the order of all the buttons so it fucks with muscle-memory chemists - those would all be cool and not magical.

**Comment by Eneocho** [comment]
Space maracas have an internal battery that powers a speaker/motion sensor, which makes the shaking sound. Emagging it damages the battery so it explodes, not magic.

**Comment by ManopolisJones** [comment]
You know it's funny, I've never actually seen anyone do a concerted e-magged maraca attack against the station. I can only imagine how effective it would be.

**Comment by BomberBro** [comment]
Somebody tell Robot about this PR

**Comment by SonixApache** [comment]
> Somebody tell Robot about this PR

it's funny how population opinions change over time
#20123

**Comment by Kurfursten** [comment]
This isn't really the same. Removal and reimagining are separate beasts to emojis.

File: code/game/objects/items/devices/maracas.dm [hunk 1]
```diff
@@ -25,9 +26,13 @@
 	chickchicky()
 
 /obj/item/device/maracas/throw_impact(atom/hit_atom, var/speed, var/mob/user)
-	if(emagged)
-		explosion(get_turf(src), -1 ,1, 3, whodunnit = user)
-		qdel(src)
+	if(!triggered)
```
Comment (Kurfursten) [inline code/game/objects/items/devices/maracas.dm]: just merge these 2 if-statements into a single line

File: code/game/objects/items/devices/maracas.dm [hunk 2]
```diff
@@ -25,9 +26,13 @@
...
+		if(emagged)
+			playsound(src, 'sound/misc/cuban_pete.ogg', 100, 0, falloff = 2)
+			triggered = 1
```
Comment (Kurfursten) [inline code/game/objects/items/devices/maracas.dm]: prefer TRUE

File: code/game/objects/items/devices/maracas.dm [hunk 3]
```diff
@@ -25,9 +26,13 @@
...
+			spawn(21) //The point in the audio file in which the tune changes
```
Comment (Kurfursten) [inline code/game/objects/items/devices/maracas.dm]: 2.1 SECONDS

**Review by Kurfursten — CHANGES_REQUESTED** [review]
those things I said

**Comment by calzilla1** [comment]
Could the ones at Centcomm still be the old ones?

**Comment by Exxion** [comment]
>Space maracas have an internal battery that powers a speaker/motion sensor, which makes the shaking sound. Emagging it damages the battery so it explodes, not magic.

This was an explanation I came up with as a joke which someone else then put in the game

**Comment by boy2mantwicethefam** [comment]
@calzilla1 They could but I'm burned out on coding.

**Comment by Exxion** [comment]
Speaking of that joke explanation, it's not really necessary anymore if they're not emaggable

File: code/game/objects/items/devices/maracas.dm (hunk 3 above)
Comment (Exxion) [reply code/game/objects/items/devices/maracas.dm]: I don't think this really matters but it's already done

**Review by Kurfursten — APPROVED** [review]

**Pull request merged**